source/Octopus.Server.Client/Exceptions/OctopusExceptionFactory.cs

```diff
@@ -51,74 +51,95 @@ public static async Task<OctopusException> CreateException(HttpResponseMessage r
 
         public static OctopusException CreateException(int statusCode, string body)
         {
-            if (statusCode == 400 || statusCode == 409) // Bad request: usually validation error 
+            if (statusCode is 400 or 409) // Bad request: usually validation error 
             {
-                var errors = JsonConvert.DeserializeObject<OctopusErrorsContract>(body);
-                return new OctopusValidationException(statusCode, errors.ErrorMessage, errors.Errors, errors.Details)
-                {
-                    HelpText = errors.HelpText
-                };
+                return CreateOctopusValidationException(statusCode, body);
             }
 
-            if (statusCode == 401 || statusCode == 403) // Forbidden, usually no API key or permissions
+            if (statusCode is 401 or 403) // Forbidden, usually no API key or permissions
             {
-                var errors = JsonConvert.DeserializeObject<OctopusErrorsContract>(body);
-                errors = errors ?? new OctopusErrorsContract
-                         {
-                             ErrorMessage =
-                                 $"The server returned a status code of {statusCode}: {body}"
```

Comment:
Refactored this error message. `OctopusSecurityException` contains the `statusCode`, having it repeated on the message is redundant. See [CreateOctopusSecurityException](https://github.com/OctopusDeploy/OctopusClients/pull/719/files#diff-e162142f5091b207e006db7032f222b1291bdbf6eb34d3bb761f4869dbc90fa6R77) and [OctopusErrorsContractFromBody](https://github.com/OctopusDeploy/OctopusClients/pull/719/files#diff-e162142f5091b207e006db7032f222b1291bdbf6eb34d3bb761f4869dbc90fa6R117)